lhotse/supervision.py

```diff
@@ -1,14 +1,68 @@
 import logging
 from dataclasses import dataclass
 from itertools import islice
-from typing import Any, Callable, Dict, Iterable, List, Mapping, Optional, Sequence
+from typing import Any, Callable, Dict, Iterable, List, Mapping, NamedTuple, Optional, Sequence
 
 from lhotse.serialization import Serializable
 from lhotse.utils import Seconds, asdict_nonull, exactly_one_not_null, fastcopy, \
     ifnone, index_by_id_and_check, \
     perturb_num_samples, split_sequence
 
 
+@dataclass(frozen=True, unsafe_hash=True)
```

Comment:
when it's derived from a NamedTuple, the dataclass part is not needed (I think it would actually be in conflict)